_posts/2024-01-22-Python-Memory-Management-and-Troubleshooting.md

```diff
@@ -0,0 +1,568 @@
+---
+title: "Tip of the Month: Python Memory Management and Troubleshooting"
+author: dave-bunten
+tags:
+  - tip-of-the-month
+  - python
+  - memory
+  - memory-management
+  - memory-allocators
+---
+
+# Tip of the Week: Python Memory Management and Troubleshooting
+
+{% include tip-of-the-month-intro.html %}
+
+## Introduction
+
+<!-- excerpt start -->
+Have you ever run Python code only to find it taking _forever_ to complete or sometime abruptly ending with an error like: `123456 Killed` or `killed (program exited with code: 137)`?
+You may have experienced memory resource or management challenges associated with these scenarios.
+This post will cover some computer memory definitions, how Python makes use of computer memory, and share some tools which may help with these types of challenges.
+<!-- excerpt end -->
+
+## What is Memory?
+
+### Computer Memory
+
+<!-- set a max width for mermaid diagram below so it doesn't render so large -->
+<style>
+.smaller-mermaid-diagram img { max-width: 500px; }
+</style>
+
+<div class="smaller-mermaid-diagram">
+
+```mermaid!
+flowchart LR
+
+subgraph computer ["Computer (resources)"]
+  memory["Memory"]
+  storage["Data Storage"]
+  cpu["CPU(s)"]
+end
+
+style computer fill:#fff,stroke:#333
+style storage fill:#fff,stroke:#333
+style cpu fill:#fff,stroke:#333
+style memory fill:#86EFAC,stroke:#333
+
+```
+
+</div>
+
+_Computer memory is a type of computer resource available for use by software on a computer_
+
+Computer memory, also sometimes known as "RAM" or "random-access memory", or "dynamic memory" is a type of resource used by computer software on a computer.
+"Computer memory stores information, such as data and programs for immediate use in the computer. ... Main memory operates at a high speed compared to non-memory storage which is slower but less expensive and oftentimes higher in capacity. " ([Wikipedia: Computer memory](https://en.wikipedia.org/wiki/Computer_memory)).
```

Comment:
Maybe _italic_ or 

>blockquote here?

Reply:
Great point, thanks I'll get this changed!